test/all.js

```diff
@@ -413,3 +413,26 @@ test('file server does not wait for reqs to finish before closing', async t => {
 
   t.is(released, 1)
 })
+
+test('get hook returning different than null still calls release hook', async function (t) {
+  t.plan(5)
+
+  const drive = tmpHyperdrive(t)
+  await drive.put('/file.txt', 'abc')
+
+  const serve = tmpServe(t, {
+    get ({ key, filename, version }) {
+      t.pass()
+      return 0
+    },
+    release ({ key, drive }) {
+      t.is(key, null)
+      t.is(drive, 0)
```

Comment:
I added this test to show what happens when not returning `null` but still falsy value, I think it happens on main too